### ISSUE TITLE
Elasticsearch service hangs in stopping status with ReadonlyREST plugin

### ISSUE THREAD

**Issue by askids** [elastic/elasticsearch#26483]
**Elasticsearch version** (`bin/elasticsearch --version`): 5.5.1

**Plugins installed**: [xpack, readonlyrest]

**JVM version** (`java -version`): 
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

**OS version** (`uname -a` if on a Unix-like system): Windows 2008 R2 Enterprise SP1

**Description of the problem including expected versus actual behavior**:
I have Elasticsearch 5.5.1 running as a windows service (this was created manually using the zip file) on Win 2008 R2 server. When I install ReadonlyREST 1.16.10_es5.5.1 (free version) and restart the service for the 1st time, it restarts properly and plugin also work (i get the prompted for id/pwd). But when I make any changes in elasticsearch.yml file and try restarting the service, the service hangs in stopping status.

I have copied the additional entries that I have on my elasticsearch.yml file. If I see the Elasticsearch log, it looks like the service stopped properly. But in the Windows service manager, the service status is shown as stopping. In the task manager, I can still see elasticsearch exe in running status.

Currently, only way to get it back to working is to restart the machine, which is definitely not something that we want to do everytime we change something on elasticsearch.yml file. If I remove the plugin, restart service works fine in the service manager.

Please let me know, if you need any further details.

**Steps to reproduce**:
1. Install 5.5.1 using zip. create service manually using SC command.
2. Install xpack 5.5.1. Enable only monitoring plugin, disable security, graph, watcher.
3. Install readonlyrest 1.16.10_es5.5.1. Update elasticsearch.yml with basic rule for Kibana (given below)
4. Restart windows service from service manager (till here it works fine. Try opening the Elasticsearch endpoint using https address. It will prompt for id/pwd. On providing the input, it will show the cluster name, version number etc)
5. Modify a comment line to elasticsearch.yml file and try restarting the Elasticsearch windows service.

```
action.destructive_requires_name: true
transport.tcp.port: 12300
action.auto_create_index: .security,.monitoring*,.watches,.triggered_watches,.watcher-history*,.kibana*
xpack.security.enabled: false
xpack.monitoring.enabled: true
xpack.graph.enabled: false
xpack.watcher.enabled: false
indices.query.bool.max_clause_count: 20000
#
# ReadonlyREST
http.type: ssl_netty4
readonlyrest:
    enable: true 
    response_if_req_forbidden: Forbidden by ReadonlyREST ES plugin
    
    ssl:
      enable: true
      # put the keystore in the same dir with elasticsearch.yml 
      keystore_file: "plugins/readonlyrest/mykeystore.jks"
      keystore_pass: Password1
      key_pass: Password1
   
    access_control_rules:

    - name: Kibana Server
      type: allow
      auth_key: kibana:kibana
      verbosity: error
```

**Provide logs (if relevant)**:
    [2017-09-03T14:28:33,721][INFO ][o.e.n.Node               ] [ESPOC2-node1] stopping ...
    [2017-09-03T14:28:33,742][INFO ][o.e.x.m.j.p.l.CppLogMessageHandler] [controller/2728] [Main.cc@168] Ml controller exiting
    [2017-09-03T14:28:33,746][INFO ][o.e.x.m.j.p.NativeController] Native controller process has stopped - no new native processes can be started
    [2017-09-03T14:28:33,815][INFO ][o.e.n.Node               ] [ESPOC2-node1] stopped
    [2017-09-03T14:28:33,816][INFO ][o.e.n.Node               ] [ESPOC2-node1] closing ...
    [2017-09-03T14:28:33,825][INFO ][o.e.n.Node               ] [ESPOC2-node1] closed

**Comment by jimczi** [comment]
The `readonlyrest` plugin is not provided by Elasticsearch and your issue seems solved when you remove it so this issue should be tracked in the appropriate repo. I see that you already opened an issue in https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/issues/265 so I hope you don't mind if I close this one. We reserved github for confirmed bugs, please reopen if the investigation on https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/issues/265 reveals a problem in ES.

**Comment by askids** [comment]
@jimczi I understand that ROR is not from Elastic. But shouldn't this be treated as a generic ES problem, since ES is writing to log as node was stopped and closed, but not really doing either of it i.e neither stopping the process nor the service? In order words, it's incorrectly determining to have stopped/closed the Elasticsearch process. Am I am missing something here?

**Comment by jimczi** [comment]
>  If I remove the plugin, restart service works fine in the service manager.

I closed this issue because we reserve github for confirmed bugs and feature requests so the first step is to determine what ROR is doing to reach this state. Removing the plugin fixed the issue, this is a clear hint that the investigation should be done on the plugin side first. I am not saying that the issue cannot be due to es but we need to be sure that this plugin is not the real cause which is why I advised you to open an issue on the plugin repo first.
